UPGRADE-API-1.12.md

```diff
@@ -225,3 +225,176 @@ The list of changed contexts:
+
+ -   public static function custom(string $url, string $method, array $additionalHeaders = [], ?string $token = null): self;
+ ```
+
+All of them have been moved to new `Sylius/Behat/Client/RequestFactory` class.
+
+#### Changes in ApiPlatformClient
+
+Followed by this change the constructor of `Sylius/Behat/Client/ApiPlatformClient` also changed:
+
+ ```diff
+     public function __construct(
+            private AbstractBrowser $client,
+            private SharedStorageInterface $sharedStorage,
+ +          private RequestFactoryInterface $requestFactory,
+            ...
+        ) {
+        }
+ ```
+
+Now we are calling method by parameter `$this->requestFactory` instead of calling `Request` class itself.
+You can see the difference of usage below:
+
+ ```diff
+     public function index(string $resource): Response
+    {
+ -      $this->request = Request::index($this->section, $resource, $this->authorizationHeader, $this->getToken());
+ +      $this->request = $this->requestFactory->index($this->section, $resource, $this->authorizationHeader, $this->getToken());
+
+        return $this->request($this->request);
+    }
+ ```
+
+The `public function buildUploadRequest(string $resource): void;` method has been removed, as it's replaced by methods in `Sylius/Behat/Client/RequestBuilder`.
+Example change of usage in `Sylius/Behat/Context/Api/Admin/ManagingAdministratorsContext`:
+
+ ```diff
+ -       $this->client->buildUploadRequest(Resources::AVATAR_IMAGES);
+ -       $this->client->addParameter('owner', $this->iriConverter->getIriFromItem($administrator));
+ -       $this->client->addFile('file', new UploadedFile($this->minkParameters['files_path'] . $avatar, basename($avatar)));
+ -       $response = $this->client->upload();
+        
+ +       $builder = RequestBuilder::create(
+ +           sprintf('/api/v2/%s/%s', 'admin', Resources::AVATAR_IMAGES),
+ +           Request::METHOD_POST,
+ +       );
+ +       $builder->withHeader('CONTENT_TYPE', 'multipart/form-data');
+ +       $builder->withHeader('HTTP_ACCEPT', 'application/ld+json');
+ +       $builder->withHeader('HTTP_Authorization', 'Bearer ' . $this->sharedStorage->get('token'));
+ +       $builder->withParameter('owner', $this->iriConverter->getIriFromItem($administrator));
+ +       $builder->withFile('file', new UploadedFile($this->minkParameters['files_path'] . $avatar, basename($avatar)));
+
+ +       $response = $this->client->request($builder->build());
+ ```
+
+As you can see the builder contains the methods that makes the request responsive for additional headers and parameters.
+
+We also changed the `request(RequestInterface $request): Response` method visibility to public.
+
+ ```diff
+ -    private function request(RequestInterface $request): Response
+ +    public function request(RequestInterface $request): Response
+ ```
+
+#### Defining ContentType changed
+
+`Sylius/Behat/Client/ContentTypeGuide` is an extended alternative for `resolveHttpRequest()` method.
```

Comment:
What `resolveHttpRequest()` is? I cannot find it anywhere 😕

Reply:
My bad, wrong name - should be resolveHttpMethod() instead 😅